### PR TITLE
Pin dockerfile version.

### PR DESCRIPTION
**Pull request by godofredoc**
This is part of the security effort to pin all the docker dependencies
across all the flutter repositories.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: ci/docker/build/Dockerfile [hunk 1]
```diff
@@ -59,10 +59,8 @@ RUN apt-get install -y \
     mtools             \
     nasm               \
     net-tools          \
-    openjdk-8-jdk      \
```
Comment (zanderso) [inline ci/docker/build/Dockerfile:62]: Why are these being removed? Are they no longer needed, or do they appear in the new image by default?

Comment (godofredoc) [reply ci/docker/build/Dockerfile:62]: They do not exist on the default packages anymore and seem like they are not needed. Waiting for the cirrus test to complete to validate they are not needed anymore.

Comment (godofredoc) [reply ci/docker/build/Dockerfile:62]: Confirmed that they are not needed anymore.

File: ci/docker/build/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM debian:stretch
+FROM debian:bullseye-slim@sha256:82da53aa627b9d5032a1e57903356b8f34d613a5bc1e07ae5e9149bd88fa3128
```
[the remark this answers is not in the document]
Comment (zanderso) [reply ci/docker/build/Dockerfile:1]: @godofredoc friendly ping on this question.

[the remark this answers is not in the document]
Comment (godofredoc) [reply ci/docker/build/Dockerfile:1]: Sorry somehow I didn't see this question. LUCI is still on stretch, I can get it back to stretch with the pinned version or we can be proactive and use the Ubuntu version Dart is currently using and that we are planning to migrate to. What option do you think will work better?

[the remark this answers is not in the document]
Comment (zanderso) [reply ci/docker/build/Dockerfile:1]: Can this file take comments, or is there a readme in this directory that can be updated to explain what's going on? Being proactive sgtm, but how will this get updated to match in the future?

[the remark this answers is not in the document]
Comment (godofredoc) [reply ci/docker/build/Dockerfile:1]: Yes, it supports comments. As long as they are on the same major version the minor updates will be applied by dependabot in this file and by `apt-get upgrade` on the VMs.

**Comment by godofredoc** [comment]
@zanderso config file updated to use ubuntu:bionic and it is ready for review.

**Review by zanderso — APPROVED** [review]

**Pull request merged**